### PR TITLE
Enable impact analysis for product tests only on PRs

### PR DESCRIPTION
**Pull request by nineinchnick**
## Description

Impact analysis for product tests is enabled on the master branch, where all tests should always run. This happens even if GIB is disabled, because we make sure the `gib-impacted.log` file always exists, even on the master branch. An empty list of impacted modules is processed by the plugin reader and produces an empty list of impacted plugins, causing most product tests to be skipped.

This PR explicitly enabled impact analysis only on PRs in forks.

> Is this change a fix, improvement, new feature, refactoring, or other?
fix

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)
CI product tests

> How would you describe this change to a non-technical end user or system administrator?
n/a

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -719,7 +719,12 @@ jobs:
         run: |
           find . -type f -name \*-executable.jar -exec chmod 0777 {} \;
       - name: Enable impact analysis
-        if: steps.filter.outputs.product-tests == 'false' && !contains(github.event.pull_request.labels.*.name, 'tests:all') && !contains(github.event.pull_request.labels.*.name, 'tests:all-product')
+        if: |
+          steps.filter.outputs.product-tests == 'false' &&
+          github.event_name == 'pull_request' &&
+          github.event.pull_request.head.repo.full_name != github.repository &&
```
Comment (losipiuk) [inline .github/workflows/ci.yml:725]: Is it "do not run impact-analyis from PR coming from the origin"?
Why so?

Comment (losipiuk) [reply .github/workflows/ci.yml:725]: I think logic is ok. But please add a comment maybe.

Comment (nineinchnick) [reply .github/workflows/ci.yml:728]: PTAL

**Review by losipiuk — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (losipiuk) [reply .github/workflows/ci.yml:728]: awesome!

**Pull request merged**